src/main/java/com/esri/samples/scene/open_mobile_scene_package/OpenMobileScenePackageSample.java

```diff
@@ -0,0 +1,148 @@
+import com.esri.arcgisruntime.mapping.view.SceneView;
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Alert;
+import javafx.scene.layout.StackPane;
+import javafx.stage.Stage;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.ExecutionException;
+
+public class OpenMobileScenePackageSample extends Application {
+
+  private SceneView sceneView;
+
+  @Override
+  public void start(Stage stage) throws Exception {
+
+    // create stack pane and application scene
+    StackPane stackPane = new StackPane();
+    Scene fxScene = new Scene(stackPane);
+
+    // set title, size, and add scene to stage
+    stage.setTitle("Open Mobile Map Package Sample");
+    stage.setWidth(800);
+    stage.setHeight(700);
+    stage.setScene(fxScene);
+    stage.show();
+
+    // create a new ArcGIS scene and set its basemap
+    ArcGISScene scene = new ArcGISScene();
+    scene.setBasemap(Basemap.createImagery());
+
+    // create a new scene view
+    sceneView = new SceneView();
+
+    // add base surface for elevation data
+    Surface surface = new Surface();
+    ArcGISTiledElevationSource elevationSource = new ArcGISTiledElevationSource("http://elevation3d.arcgis" +
+            ".com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer");
+    surface.getElevationSources().add(elevationSource);
+    scene.setBaseSurface(surface);
+
+    // load a mobile scene package
+    final String mspkPath = new File("./samples-data/mspk/philadelphia.mspk").getAbsolutePath();
+    // create a temporary directory to store unpacked file if appropriate
+    Path tempDirectory = Files.createTempDirectory("offline_map");
+    final String tempUnpackedPath = tempDirectory.toString();
+
+    //instantiate mobile scene packages for direct read and unpacked options
+    MobileScenePackage directReadMSPK = new MobileScenePackage(mspkPath);
+    MobileScenePackage unpackedMSPK = new MobileScenePackage(tempUnpackedPath);
+
+    // check if the mobile scene package can be read directly using a static method
+    ListenableFuture<Boolean> isDirectReadSupported = MobileScenePackage.isDirectReadSupportedAsync(mspkPath);
+    isDirectReadSupported.addDoneListener(() -> {
+
+      try {
+        // if the mobile scene package can be read directly, then load the mspk from the direct read path directory
+        if (isDirectReadSupported.get()) {
+          loadMobileScenePackage(directReadMSPK);
+
+        } else {
+          // if the mobile scene package has to be unpacked, then unpack the mobile scene package and store it in a local path
+          MobileScenePackage.unpackAsync(mspkPath, tempUnpackedPath).addDoneListener(() -> {
+            // load the mobile scene package from the unpacked path
+            loadMobileScenePackage(unpackedMSPK);
+          });
+        }
+
+      } catch (InterruptedException | ExecutionException e) {
+        e.printStackTrace();
```

Comment:
Should show an alert here with something like "mobile scene package direct read could not be determined"